includes/fields/class-color.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Color field.
+ *
+ * @package HivePress\Fields
+ */
+
+namespace HivePress\Fields;
+
+use HivePress\Helpers as hp;
+
+// Exit if accessed directly.
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Color.
+ */
+class Color extends Text {
+	/**
+	 * Class initializer.
+	 *
+	 * @param array $meta Class meta values.
+	 */
+	public static function init( $meta = [] ) {
+		$meta = hp\merge_arrays(
+			[
+				'label'    => esc_html__( 'Color', 'hivepress' ),
+				'sortable' => false,
+
+				'settings' => [
+					'min_length' => null,
+					'max_length' => null,
+				],
+			],
+			$meta
+		);
+
+		parent::init( $meta );
+	}
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param array $args Field arguments.
+	 */
+	public function __construct( $args = [] ) {
+		$args = hp\merge_arrays(
+			$args,
+			[
+				'pattern'      => '^#([A-Fa-f0-9]{3}){1,2}$',
+				'display_type' => 'text',
+			]
+		);
+
+		parent::__construct( $args );
+	}
+
+	/**
+	 * Sanitizes field value.
+	 */
+	protected function sanitize() {
+		$this->value = sanitize_hex_color( $this->value );
+	}
+
+	/**
+	 * Renders field HTML.
+	 *
+	 * @return string
+	 */
+	public function render() {
+		return '<input data-component="color-picker" type="' . esc_attr( $this->display_type ) . '" name="' . esc_attr( $this->name ) . '" value="' . esc_attr( $this->value ) . '" ' . hp\html_attributes( $this->attributes ) . '>';
```

Comment:
Is overriding the whole render method required? If the reason is "data-component" then it can be added to the attributes array in the boot method instead.